src/blue.ts

```diff
@@ -224,8 +212,82 @@ export function blueProxyFactory(env: MembraneBroker) {
             }
             return env.getBlueValue(red);
         }
+        /**
+         * This trap cannot just use `Reflect.get` directly on the `target` because
+         * the red object graph might have mutations that are only visible on the red side,
+         * which means looking into `target` directly is not viable. Instead, we need to
+         * implement a more crafty solution that looks into target's own properties, or
+         * in the red proto chain when needed.
+         */
+        get(shadowTarget: BlueShadowTarget, key: PropertyKey, receiver: BlueObject): RedValue {
+            /**
+             * If the target has a non-configurable own data descriptor that was observed by the red side,
+             * and therefore installed in the shadowTarget, we might get into a situation where a writable,
+             * non-configurable value in the target is out of sync with the shadowTarget's value for the same
+             * key. This is fine because this does not violate the object invariants, and even though they
+             * are out of sync, the original descriptor can only change to something that is compatible with
+             * what was installed in shadowTarget, and in order to observe that, the getOwnPropertyDescriptor
+             * trap must be used, which will take care of synchronizing them again.
+             */
+            const { target } = this;
+            const redDescriptor = ReflectGetOwnPropertyDescriptor(target, key);
+            if (isUndefined(redDescriptor)) {
+                // looking in the blue proto chain to avoid switching sides
+                const blueProto = getBlueValue(ReflectGetPrototypeOf(target));
```

Comment:
The follow-up PR will handle cases where the prototype value is `null`.

Reply:
yes, coming!